draft-ietf-rats-corim.md

```diff
@@ -2070,82 +2042,147 @@ Logically, new ECT entries are appended to the existing ACS.
+A Verifier SHALL create a "matched entries" set, and SHALL populate it with all ACS entries which matched the condition ECT.
+
+If the matched entries array is not empty, then the condition ECT matches the ACS.
+
+If the matched entries array is empty, then the condition ECT does not match the ACS.
+
+### Comparing a condition ECT against a single ACS entry {#sec-match-one-condition-ect}
 
-### Environment Comparison {#sec-compare-env}
+A Verifier SHALL perform all of the comparisons in sections {{sec-compare-environment}), {{sec-compare-authority}), {{sec-compare-cm}} and {{sec-compare-element-list}).
+Each of these comparisons compares one field in the condition ECT against the same field in the ACS entry.
 
-### Claims Comparison  {#sec-compare-claims}
+If all of the fields match, then the condition ECT matches the ACS entry.
 
-#### Comparison of measurement-values-map {#sec-match-one-codepoint}
+If any of the fields does not match, then the condition ECT does not match the ACS entry.
+
+### Environment Comparison {#sec-compare-environment}
+
+A Verifier SHALL compare each field which is present in the condition ECT `environment-map` against the corresponding field in the ACS entry `environment-map` using binary comparison.
+Before performing the binary comparison, a Verifier SHOULD convert both `environment-map` fields into a form which meets CBOR Core Deterministic Encoding Requirements {{-cbor}}.
+
+If all fields which are present in the condition ECT `environment-map` are present in the ACS entry and are binary identical, then the environments match.
+
+If any field which is present in the condition ECT `environment-map` is not present in the ACS entry, then the environments do not match.
+
+If any field which is present in the condition ECT `environment-map` is not binary identical to the corresponding ACS entry field, then the environments do not match.
+
+If a field is not present in the condition ECT `environment-map` then the presence of, and value of, the corresponding ACS entry field SHALL NOT affect whether the environments match.
+
+### Authority comparison {#sec-compare-authority}
+
+A Verifier SHALL compare the condition ECT expected `authority` value to the candidate entry `authority` value.
+
+If every entry in the condition ECT `authority` has a matching entry in the ACS entry `authority` field, then the authorities match.
+The order of the fields in each `authority` field do not affect the result of the comparison.
+
+If any entry in the condition ECT `authority` does not have a matching entry in the ACS entry `authority` field then the authorities do not match.
+
+When comparing two `$crypto-key-type-choice` fields for equality, a Verifier SHALL treat them as equal if their deterministic CBOR encoding is binary equal.
+
+
+
+### Conceptual Message Type Comparison  {#sec-compare-cm}
```

Comment:
There isn't an external representation for cm-type in comid. Hence, there isn't a way to include it in a `condition` statement. The semantics for cm-type are intrinsic to processing semantics of CMs. For example, processing a RV message means the Evidence ECTs are the target ACS entries. For endorsements, Evidence and RV ECTs are the targets. Note: There is an unresolved issue as to whether Endorsement conditions can target other Endorsement ECTs.

Consequently, I don't believe this section is needed. The semantics of cm-type will be handled in the sections that describe processing phases.

Reply:
Ok, I will remove it.